### PR TITLE
Don't self-use deprecated API

### PR DESCRIPTION
**Pull request by Carreau**
Need to dig back a few version numbers, and update a few deprecation warnings as well .

### CODE REVIEW

File: ipykernel/pickleutil.py [hunk 1]
```diff
@@ -21,9 +20,13 @@
 from ipython_genutils.importstring import import_item
 from ipython_genutils.py3compat import string_types, iteritems, buffer_to_bytes, buffer_to_bytes_py2
 
-from . import codeutil  # This registers a hook when it's imported
+# This registers a hook when it's imported
+try:
+    from ipyparallel.serialize import codeutil
+except ImportError:
+    # Deprecated since <value>....
```
Comment (takluyver) [inline ipykernel/pickleutil.py]: `<value>` was such a great release

Comment (Carreau) [inline ipykernel/pickleutil.py]: Yeah, I know, but 9h ago was 2h too late. We miss you.

Comment (takluyver) [inline ipykernel/pickleutil.py]: Miss you too. Especially when you're all swanning around in Austin ;-)

Comment (Carreau) [inline ipykernel/pickleutil.py]: Took me 24 days, but now a version has a number.

File: ipykernel/codeutil.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 # Distributed under the terms of the Modified BSD License.
 
 import warnings
-warnings.warn("ipykernel.codeutil is deprecated. It has moved to ipyparallel.serialize", DeprecationWarning)
+warnings.warn("ipykernel.codeutil is deprecated since IPytkernel 4.3.1. It has moved to ipyparallel.serialize", DeprecationWarning)
```
Comment (takluyver) [inline ipykernel/codeutil.py]: IPytkernel -> IPykernel

Comment (Carreau) [inline ipykernel/codeutil.py]: Fixed.

**Pull request merged**